dist/tools/ci/print_toolchain_versions.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```

Comment:
unless there's a good reason, please revert this.

Reply:
`/bin/sh` has different behavior than `/bin/bash` and some distributions, like Arch Linux symlink `/bin/sh` to `/bin/bash` by default it means you could get different behavior without even noticing on different systems.

By being explicit it ensures you do not get into these maintenance troubles.

Reply:
If a script is using bash specifics, use ```/bin/bash```. But *try to avoid that*.

POSIX plus some extensions are just fine and what we've been using all along.

Reply:
```read -r osname osvers <<<$(get_windows_info)```

The `<<<` is bash specific.

Reply:
can be achieved with

```sh
get_windows_info | read -r osname osvers
```

without bashisms.

Reply:
No, the `read` is started on a subshell and the variables it creates die with that subshell. See [this question](https://stackoverflow.com/questions/2746553/read-values-into-a-shell-variable-from-a-pipe) for a discussion of this issue.

Reply:
If there is really an objection to use `bash`, a solution could be to do two calls instead of one, like done for `Linux`.

Reply:
Ah, how about

```sh
local info = $(get_windows_info)
echo "$info" | cut -d' ' -f1
echo "$info" | cut -d' ' -f2-
```

then?

> If there is really an objection to use bash, a solution could be to do two calls instead of one, like done for Linux.

Not everyone has `bash` installed or can (think slim OSs like OpenWrt e.g.) or wants to.

Reply:
It is already required for many things:

```
git grep -e '/bin/bash' -e '/usr/bin/env bash'
boards/common/arduino-atmega/dist/debug.sh:#!/usr/bin/env bash
boards/common/arduino-atmega/dist/debug_srv.sh:#!/usr/bin/env bash
boards/common/msba2/tools/flashutil.sh:#/usr/bin/env bash
boards/mega-xplained/dist/debug.sh:#!/usr/bin/env bash
boards/mega-xplained/dist/debug_srv.sh:#!/usr/bin/env bash
boards/waspmote-pro/dist/debug.sh:#!/usr/bin/env bash
boards/waspmote-pro/dist/debug_srv.sh:#!/usr/bin/env bash
dist/tools/buildsystem_sanity_check/check.sh:#!/usr/bin/env bash
dist/tools/ci/build_and_test.sh:#!/usr/bin/env bash
dist/tools/doccheck/check.sh:#!/usr/bin/env bash
dist/tools/eclipsesym/cmdline2xml.sh:#!/usr/bin/env bash
dist/tools/flake8/check.sh:#!/usr/bin/env bash
dist/tools/has_minimal_version/has_minimal_version.sh:#! /bin/bash
dist/tools/openocd/openocd.sh:#!/bin/bash
dist/tools/packer/setup.sh:#!/bin/bash -eux
dist/tools/pr_check/check_labels.sh:#!/usr/bin/env bash
dist/tools/pr_check/pr_check.sh:#!/usr/bin/env bash
dist/tools/shellcheck/check.sh:#!/usr/bin/env bash
dist/tools/static-tests.sh:#!/bin/bash
dist/tools/toolchains/build_gnuarm.sh:#!/usr/bin/env bash
dist/tools/toolchains/build_gnuarm_naked.sh:#!/usr/bin/env bash
dist/tools/usb-serial/find-tty.sh:#!/usr/bin/env bash
dist/tools/vagrant/bootstrap.sh:#!/usr/bin/env bash
Binary file pkg/oonf_api/patches/0002-port-tests-to-riot.patch matches
tests/pkg_fatfs/create_fat_image_file.sh:#!/usr/bin/env bash
tests/pkg_fatfs_vfs/create_fat_image_file.sh:#!/usr/bin/env bash
```

Reply:
As always the question: should we make matters worse? Also, IMHO it should not be required by such a basic tool that [we explicitly ask users to call when they open an issue](https://github.com/RIOT-OS/RIOT/blame/6e9612f1d1e41ef56b79c1fcddc3d7e3aae2b9d7/.github/ISSUE_TEMPLATE/bug_report.md#L44), especially if there are solutions to do it without bash-isms.

Reply:
I don't think we are are making matters "worse" in the sense that we already established we want to use bash because we need the features it offers. I tend to like pure vanilla SH, but the decision to go bash has already been taken.

Now, I understand that the purpose of this script is a tiny bit defeated by requiring bash. It would be unable to detect that you have an unsupported shell, though you would see that when your command line tells you "bash: command not found".

Again, BASH is already the standard in RIOT.

Reply:
But why introduce bash-isms in this instance if they are not necessary?